app/assets/javascripts/models/beta/iteration.js

```diff
@@ -1,21 +1,129 @@
-import moment from 'moment';
+import moment from "moment";
+import { status, storyTypes } from "libs/beta/constants";
+import * as Story from "./story";
 
 const weeksBetween = (dateA, dateB) =>
-  moment(new Date(dateA)).diff(moment(new Date(dateB)), 'week');
+  moment(new Date(dateA)).diff(moment(new Date(dateB)), "week");
 
 const getIterationForDate = (date, project) => {
   const weeks = weeksBetween(date, project.startDate);
   const iterationQuantity = Math.ceil(weeks / project.iterationLength);
 
-  return project.iterationLength > weeks ? iterationQuantity : iterationQuantity + 1;
+  return project.iterationLength > weeks
+    ? iterationQuantity
+    : iterationQuantity + 1;
 };
 
-export const getCurrentIteration = (project) => (
-  getIterationForDate(new Date(), project)
-);
+export const getDateForIterationNumber = (iterationNumber, project) => {
+  return moment(new Date(project.startDate))
+    .startOf("isoWeek")
+    .add(iterationNumber, "weeks")
+    .format("ddd MMM Do Y");
+};
+
+export const getCurrentIteration = project =>
+  getIterationForDate(new Date(), project);
 
 export const getIterationForStory = (story, project) => {
-  if (story.state === 'accepted') {
+  if (story.state === "accepted") {
     return getIterationForDate(new Date(story.acceptedAt), project);
   }
-}
+};
+
+const createSprint = (sprintNumber = 0, startDate = 0, isFiller = false) => ({
+  number: sprintNumber,
+  startDate: startDate,
+  points: 0,
+  completedPoints: null,
+  stories: [],
+  isFiller: isFiller
+});
+
+const createFillerSprints = (size, initialNumber, project) => {
+  let fillerSprints = [];
+  for (let i = 1; i <= size; i++) {
+    const sprintNumber = initialNumber + i;
+    const sprint = createSprint(
+      sprintNumber,
+      getDateForIterationNumber(sprintNumber, project),
+      true
+    );
+    fillerSprints.push(sprint);
+  }
+  return fillerSprints;
+};
+
+const lastRealSprintIndex = sprints => {
+  const realSprints =
+    sprints.length && sprints.filter(sprint => sprint && !sprint.isFiller);
+  const index = realSprints.length > 0 ? realSprints.length - 1 : 0;
+  return sprints.indexOf(realSprints[index]);
+};
+
+const canTakeStory = (sprint, storyPoints, velocity) => {
+  if (sprint && !sprint.isFiller) {
+    return (
+      sprint.points + storyPoints <= velocity ||
+      sprint.points === 0 ||
+      storyPoints === 0
```

Comment:
What do you think to replace this part by:

```
sprint.points + storyPoints <= velocity ||
!sprint.points ||
!storyPoints
```

and put this part inside a function

Reply:
perhaps you mean double bangs, like `!!sprint.points` right?

Reply:
Do you think that it is really necessary to create another function for a conditional that has the same meaning as the parent function?

I mean, `canTakeStory` has only 5 lines e the sole purpose of that function is to return that conditional....

Reply:
In my point of view, yes. Not because of the number of lines but because of the meaning. For example:

```
const canTakeStory = (sprint, storyPoints, velocity) => sprint.points + storyPoints <= velocity || !sprint.points || !storyPoints;
```

and

```
return canTakeStory(sprint, storyPoints, velocity);
```

I think it is better.